src/ng/q.js

```diff
@@ -243,7 +243,9 @@ function $$QProvider() {
  */
 function qFactory(nextTick, exceptionHandler) {
   var $qMinErr = minErr('$q', TypeError);
-
+  window.promises = window.promises || {};
```

Comment:
Remove this line and the line below, encapsulate it away in the implementation of the trackPromise, untrackPromise